lib/trace.c

```diff
@@ -148,18 +148,23 @@ M0_INTERNAL void m0_trace_fini(void)
 	m0_arch_trace_fini();
 }
 
-
-/*
- * XXX x86_64 version.
- */
 static inline uint64_t m0_rdtsc(void)
 {
+#ifdef CONFIG_X86_64
 	uint32_t count_hi;
 	uint32_t count_lo;
 
 	__asm__ __volatile__("rdtsc" : "=a"(count_lo), "=d"(count_hi));
 
 	return ((uint64_t)count_lo) | (((uint64_t)count_hi) << 32);
+#elif defined CONFIG_AARCH64
+	uint64_t cycle;
+
+	asm volatile("mrs %0, pmccntr_el0" : "=r"(cycle));
```

Comment:
Will this work, if kernel component is not there for AARCH64 ?

Reply:
No, it is framed by  platform specific macro

Reply:
Can you add a todo here, in case of user mode only support needed for AARCH, a different mechanism is needed. ?

Reply:
Added